backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/controller/StockViewController.java

```diff
@@ -469,7 +471,9 @@ private ReportedMaterialStockDto convertToDto(ReportedMaterialItemStock entity)
         @ApiResponse(responseCode = "200", description = "OK"),
         @ApiResponse(responseCode = "400", description = "Invalid parameter", content = @Content)
     })
-    public ResponseEntity<List<ReportedProductStockDto>> getCustomerProductStocks(@RequestParam String ownMaterialNumber) {
+    public ResponseEntity<List<ReportedProductStockDto>> getCustomerProductStocks(
```

Comment:
As you pointed out, the failure lies on the frontend side, because the frontend issues a call with a null value in a mandatory Request Parameter. In spite of that, I'd rather solve this by adding some error handling to the StockViewController endpoints in the backend, since I'm not very familiar with the frontend and don't want to change things there, if I can avoid it.

Reply:
See this as partially solved. Within controller we did not explicitly stated the parameter to be non-null (checking and handling this definitely is an improvement! :)).

While running the frontend currently an error comes up in the log which is not a state we should keep (as we previously didn't have an error). I also looked up how to do it accordingly.

The change needed is the following: The StockViewDetails currently uses the hook `usePartnerStocks` and hands over the materialNumberSupplier / materialNumberCustomer.  As we use optional chaining (? operator returns undefined if attribute / object is null or undefined) we might handover undefined which is not handled accordingly by the hook. Adding the nullish coalescing operator with null (?? null) returns null if the left-hand side is undefined or null. That way the fetch will not be performed in the hook :)

```Typescript
# old line 45 in StockDetailsView
    const { partnerStocks } = usePartnerStocks(
        type,
        type === 'product' ? selectedMaterial?.material?.materialNumberSupplier: selectedMaterial?.material?.materialNumberCustomer
    );

# new line 45 in StockDetailsView
    const { partnerStocks } = usePartnerStocks(
        type,
        type === 'product' ? selectedMaterial?.material?.materialNumberSupplier ?? null : selectedMaterial?.material?.materialNumberCustomer ?? null
    );
```

I committted that, but wanted to share that so that we share the insights here :)